pkg/inst-api-semconv/instrumenter/http/http_attrs_extractor.go

```diff
@@ -0,0 +1,114 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"context"
+	"github.com/open-telemetry/opentelemetry-go-compile-instrumentation/pkg/inst-api-semconv/instrumenter/utils"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.30.0"
+	"sync/atomic"
+)
+
+type HttpCommonAttrsExtractor[REQUEST any, RESPONSE any, GETTER1 HttpCommonAttrsGetter[REQUEST, RESPONSE]] struct {
```

Comment:
I'm not really fond of using type parameters this verbose.
Maybe instead we can use marker interfaces instead of `any` and keep these less verbose.